### PR TITLE
terraform upgrade to 1.5.7

### LINKED ISSUE #1913: [ENH] - Pin terraform to lattest MPL licensed version
### Feature description

Due to the recently licensing changes [FAQ license change](https://www.hashicorp.com/license-faq) we want to pin terraform for the time being the latest MPL version.

We haven't had a lawyer involved but the language is concerning:
 - https://www.hashicorp.com/license-faq#assisting-customers-to-use-products
 - https://www.hashicorp.com/license-faq#non-competitive-oss-usage
 - https://www.hashicorp.com/license-faq#what-is-considered-competitive

Also as a whole the nebari community feels that the BSL license is toxic to open-source. We are considering in the future moving towards an open source terraform fork.

### Value and/or benefit

Avoid licensing issues.

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sblair-metrostar**
Fixes #1913 

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

## Any other comments?

Only tested on AWS so far. Needs to be run in each supported cloud environment.

### CODE REVIEW

**Comment by sblair-metrostar** [comment]
There appears to be a [bug in the kubernetes backend provider](https://github.com/hashicorp/terraform/issues/31275) that affects local deployments. Trying to apply their workaround of setting the config path but if it doesn't work we'll likely want to push this to a future release. Not enough short term value to justify the time.

File: src/_nebari/stages/infrastructure/template/local/outputs.tf [hunk 1]
```diff
@@ -2,6 +2,7 @@ output "kubernetes_credentials" {
   description = "Parameters needed to connect to kubernetes cluster locally"
   sensitive   = true
   value = {
+    config_path            = var.kubeconfig_filename
```
Comment (sblair-metrostar) [inline src/_nebari/stages/infrastructure/template/local/outputs.tf:5]: Workaround for a [kubernetes backend provider bug](https://github.com/hashicorp/terraform/issues/31275)

**Comment by iameskild** [comment]
I successfully tested this branch on an existing Digital Ocean deployment. 

Side note: This version of TF seems to be significantly faster 🎉

**Comment by iameskild** [comment]
Testing this with GCP failed because of the deprecation of `module_variable_optional_attrs`. Removing the the [provider.tf ](https://github.com/nebari-dev/nebari/blob/develop/src/_nebari/stages/infrastructure/template/gcp/provider.tf) file altogether seems to resolve the issue without any known side-effects; I'm not sure why we were using it in the first place.

**Comment by iameskild** [comment]
Okay deploying this branch on Azure was also successful. This is ready to be merged 🎉

**Review by iameskild — APPROVED** [review]

**Pull request merged**